src/run.jl

```diff
@@ -1,133 +1,160 @@
-        end
-    end
+const rootfs_cache = Dict()
 
-    return
+lazy_artifact(x) = @artifact_str(x)
+
+const rootfs_lock = ReentrantLock()
+rootfs() = lock(rootfs_lock) do
+    lazy_artifact("debian")
 end
 
 """
-    run_sandboxed_julia(install::String, args=``; wait=true, interactive=true,
-                        stdin=stdin, stdout=stdout, stderr=stderr,
-                        cache=nothing, storage=nothing, kwargs...)
-
-Run Julia inside of a sandbox, passing the given arguments `args` to it. The argument
-`install` specifies the directory where Julia can be found.
-
-The argument `wait` determines if the process will be waited on, and defaults to true. If
-setting this argument to `false`, remember that the sandbox is using on Docker and killing
-the process does not necessarily kill the container. It is advised to use the `name` keyword
-argument to set the container name, and use that to kill the Julia process.
+    run_sandboxed_julia(install::String, args=``; env=Dict(), mounts=Dict(),
+                        wait=true, stdin=stdin, stdout=stdout, stderr=stderr,
+                        install_dir="/opt/julia", kwargs...)
 
-The `cache` directory is used to cache temporary files across runs, e.g. compilation caches,
-and can be expected to be removed at any point. The `storage` directory can be used for more
-lasting files, e.g. artifacts.
+Run Julia inside of a sandbox, passing the given arguments `args` to it. The argument `wait`
+determines if the process will be waited on. Streams can be connected using the `stdin`,
+`stdout` and `sterr` arguments. Returns a `Process` object.
 
-The keyword argument `interactive` maps to the Docker option, and defaults to true.
+Further customization is possible using the `env` arg, to set environment variables, and the
+`mounts` argument to mount additional directories. With `install_dir`, the directory where
+Julia is installed can be chosen.
 """
-function run_sandboxed_julia(install::String, args=``; wait=true,
-                             stdin=stdin, stdout=stdout, stderr=stderr, kwargs...)
-    cmd = runner_sandboxed_julia(install, args; kwargs...)
-    Base.run(pipeline(cmd, stdin=stdin, stdout=stdout, stderr=stderr); wait=wait)
-end
-
-function runner_sandboxed_julia(install::String, args=``; interactive=true, tty=true,
-                                name=nothing, cpus::Vector{Int}=Int[], tmpfs::Bool=true,
-                                storage=nothing, cache=nothing, sysimage=nothing,
-                                xvfb::Bool=true, init::Bool=true,
-                                runner="ubuntu", user="pkgeval", group="pkgeval",
-                                install_dir="/opt/julia")
-    ## Docker args
+function run_sandboxed_julia(install::String, args=``; wait=true, kwargs...)
+    config, cmd = runner_sandboxed_julia(install, args; kwargs...)
 
-    cmd = `docker run --rm`
+    # XXX: even when preferred_executor() returns UnprivilegedUserNamespacesExecutor,
```

Comment:
This was with the lock around the global executor cache from https://github.com/staticfloat/Sandbox.jl/pull/15